File: docusign-sdk-sample-objc/docusign-sdk-sample-objc/Controllers/LoginViewController.m
//
//  LoginViewController.m
//  docusign-sdk-sample-objc
//
//  Copyright © 2017 DocuSign. All rights reserved.
//

#import "LoginViewController.h"

#import "AppDelegate.h"
#import <DocuSignSDK/DocuSignSDK.h>
#import <ProgressHUD/ProgressHUD.h>
#import "ProfileManager.h"

@interface LoginViewController () <UITextFieldDelegate>

// ui elements
@property (nonatomic, strong) IBOutlet UITextField * usernameInput;
@property (nonatomic, strong) IBOutlet UITextField * passwordInput;

@end


@implementation LoginViewController


#pragma mark - UI Lifecycle Methods

- (void) viewDidLoad
{
    [super viewDidLoad];
   
    // pre-populate the user credentials fields with defaults (if provided)
    NSDictionary * defaultUserCredentials = [ProfileManager getDefaultUserCredentials];
    self.usernameInput.text = defaultUserCredentials[@"defaultUsername"];
    self.passwordInput.text = defaultUserCredentials[@"defaultPassword"];
    
    self.passwordInput.delegate = self;
}


- (void) viewWillAppear:(BOOL)animated
{
    // screen should only display portrait
    AppDelegate* appDelegate = (AppDelegate*)[UIApplication sharedApplication].delegate;
    appDelegate.restrictRotation = TRUE;
}


#pragma mark - IBAction Methods

- (IBAction) loginButtonTapped:(UIButton *)sender
{

    [ProgressHUD show:@"Authenticating..."];

    NSString * email = self.usernameInput.text;
    NSString * password = self.passwordInput.text;
    
    // check username and password formats are valid
    if ([self isUsernameFormatValid:email] && [self isPasswordFormatValid:password]) {
        
        // log user into docusign api
        NSString * integratorKey = [ProfileManager getIntegratorKey];
        NSString * hostApiUrl = [ProfileManager getHostApiUrl];
        [DSMManager loginWithEmail:email password:password integratorKey:integratorKey host:[NSURL URLWithString:hostApiUrl] completion:^(DSMAccountInfo *accountInfo, NSError *error) {
            
                if (error != nil) {
                    NSLog(@"An error occurred attempting to log into DocuSign: %@", error);
                    [ProgressHUD dismiss];
                    
                    // display alert to user
                    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Login Failed"
                                                                                             message:error.localizedDescription
                                                                                      preferredStyle:UIAlertControllerStyleAlert];
                    UIAlertAction *actionOk = [UIAlertAction actionWithTitle:@"Ok" style:UIAlertActionStyleDefault handler:nil];
                    [alertController addAction:actionOk];
                    [self presentViewController:alertController animated:TRUE completion:nil];
                }
                else {
                    NSLog(@"User Authenticated");
                    [ProgressHUD dismiss];

                    // proceed to app main nav
                    [self performSegueWithIdentifier:@"segueMainNav" sender:self];
                }
            }];
    }
}

- (IBAction)passwordFieldPrimaryAction:(id)sender {
    [sender resignFirstResponder];
}

- (BOOL)textFieldShouldReturn:(UITextField *)textField {
    [self.passwordInput resignFirstResponder];
    return YES;
}

#pragma mark - Private Methods

- (bool) isUsernameFormatValid:(NSString *)username
{
    // check username is non-empty
    // TODO perform additional validation
    return (username.length > 0);
}


- (bool) isPasswordFormatValid:(NSString *)password
{
    // check password is non-empty
    // TODO Perform additional validation
    return (password.length > 0);
}


@end

File: docusign-sdk-sample-swift/docusign-sdk-sample-swift/Controllers/LoginViewController.swift
//
//  LoginViewController.swift
//  docusign-sdk-sample-swift
//
//  
//  Copyright © 2017 DocuSign. All rights reserved.
//


import DocuSignSDK
import SVProgressHUD
import UIKit


class LoginViewController: UIViewController, UITextFieldDelegate
{
    @IBOutlet weak var tf_username: UITextField!
    @IBOutlet weak var tf_password: UITextField!


    // MARK: UI Lifecycle Methods

    override func viewDidLoad()
    {
        super.viewDidLoad()

        // pre-populate the user credentials fields with defaults (if provided)
        tf_username.text = ProfileManager.Static.defaultUsername
        tf_password.text = ProfileManager.Static.defaultPassword
        tf_password.delegate = self
    }
    
    // MARK: IBAction Methods

    @IBAction func signInButtonTapped(_ sender: Any)
    {
        SVProgressHUD.show(withStatus: "Authenticating...")
        
        guard let email = tf_username.text else {
            return
        }
        
        guard let password = tf_password.text else {
            return
        }
        let integratorKey = ProfileManager.Static.integratorKey
        let hostUrl: URL! = ProfileManager.Static.demoHostApi
        
        if (isUsernameFormatValid(username: email) && isPasswordFormatValid(password: password))
        {
//            DSMManager.login(withEmail: email,
//                             password: password,
//                             integratorKey: integratorKey,
//                             host: hostUrl) { (accountInfo, error) in
//                SVProgressHUD.dismiss()
//                
//                if (error != nil)
//                {
//                    NSLog("Error logging in")
//                    // display error prompt
//                    self.promptError(err: error)
//                }
//                else
//                {
//                    NSLog("User authenticated")
//                    // segue to main navigation controller
//                    self.performSegue(withIdentifier: "segueMainNav", sender: nil)
//                }
//            }
        }
    }

    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        self.tf_password.resignFirstResponder()
        return true
    }

    // MARK: Private Methods
    
    func isUsernameFormatValid(username: String?) -> Bool
    {
        // check username is non-empty
        // TODO perform additional validation
        return ((username?.count)! > 0)
    }
    
    
    func isPasswordFormatValid(password: String?) -> Bool
    {
        // check password is non-empty
        // TODO Perform additional validation
        return ((password?.count)! > 0)
    }

    
    func promptError(err: Error?)
    {
        let title = "Error"
        let message = err?.localizedDescription

        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
        alert.addAction(UIAlertAction(title: "OK", style: .default) { action in })
        
        self.present(alert, animated: true)
    }
}
